libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/gqlvalue.kt

```diff
@@ -14,7 +14,12 @@ import com.apollographql.apollo3.ast.internal.validateAndCoerceValue
  */
 fun GQLValue.coerceInExecutableContextOrThrow(expectedType: GQLType, schema: Schema): GQLValue {
   val scope = DefaultValidationScope(schema)
-  val coercedValue = scope.validateAndCoerceValue(this, expectedType, false) {
+  val coercedValue = scope.validateAndCoerceValue(
+      value = this,
+      expectedType = expectedType,
+      hasLocationDefaultValue = false,
+      isOneOfInputObject = false,
```

Comment:
I think this needs to be passed down from `SelectionSetsBuilder`. An argument may be a `@oneOfType`

Reply:
Thinking about this more, there's no need to tunnel `isOneOfInputObject` all the way down. That information is already present in `schema` + `expectedType`. "Just" lookup the `isOneOf` from the schema every time.

Reply:
Nevermind, this is the 'isOneOf' of the field and not of the type so that information is not contained in the `GQLType`